basic_autonomy_ros2/include/basic_autonomy_ros2/basic_autonomy.hpp

```diff
@@ -0,0 +1,351 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <iostream>
+#include <vector>
+#include <rclcpp/rclcpp.hpp>
+#include <carma_planning_msgs/msg/trajectory_plan.hpp>
+#include <carma_planning_msgs/msg/trajectory_plan_point.hpp>
+#include <carma_planning_msgs/msg/plugin.h>
+#include <boost/shared_ptr.hpp>
+#include <carma_ros2_utils/containers/containers.hpp>
+#include <trajectory_utils/trajectory_utils.hpp>
+#include <trajectory_utils/conversions/conversions.hpp>
+#include <boost/geometry.hpp>
+#include <carma_wm_ros2/Geometry.hpp>
+#include <carma_planning_msgs/srv/plan_trajectory.hpp>
+#include <carma_wm_ros2/WMListener.hpp>
+#include <functional>
+#include <unordered_set>
+#include <autoware_msgs/msg/lane.h>
+#include <lanelet2_core/geometry/Point.h>
+#include <basic_autonomy_ros2/smoothing/SplineI.hpp>
+#include <basic_autonomy_ros2/smoothing/BSpline.hpp>
+#include <basic_autonomy_ros2/smoothing/filters.hpp>
+#include <carma_debug_ros2_msgs/msg/trajectory_curvature_speeds.hpp>
+
+/**
+ * \brief Macro definition to enable easier access to fields shared across the maneuver types
+ * \param mvr The maneuver object to invoke the accessors on
+ * \param property The name of the field to access on the specific maneuver types. Must be shared by all extant maneuver types
+ * \return Expands to an expression (in the form of chained ternary operators) that evalutes to the desired field
+ */
+#define GET_MANEUVER_PROPERTY(mvr, property)\
+        (((mvr).type == carma_planning_msgs::msg::Maneuver::INTERSECTION_TRANSIT_LEFT_TURN ? (mvr).intersection_transit_left_turn_maneuver.property :\
+            ((mvr).type == carma_planning_msgs::msg::Maneuver::INTERSECTION_TRANSIT_RIGHT_TURN ? (mvr).intersection_transit_right_turn_maneuver.property :\
+                ((mvr).type == carma_planning_msgs::msg::Maneuver::INTERSECTION_TRANSIT_STRAIGHT ? (mvr).intersection_transit_straight_maneuver.property :\
+                    ((mvr).type == carma_planning_msgs::msg::Maneuver::LANE_CHANGE ? (mvr).lane_change_maneuver.property :\
+                        (mvr).type == carma_planning_msgs::msg::Maneuver::LANE_FOLLOWING ? (mvr).lane_following_maneuver.property :\
+                        (mvr).type == carma_planning_msgs::msg::Maneuver::STOP_AND_WAIT ? (mvr).lane_following_maneuver.property :\
+                        throw new std::invalid_argument("GET_MANEUVER_PROPERTY (property) called on maneuver with invalid type id"))))))
+
+namespace basic_autonomy
+{
+    namespace waypoint_generation
+    {
+        static const std::string BASIC_AUTONOMY_LOGGER = "basic_autonomy";
```

Comment:
Just to follow the same syntax as other packages, can we give define this as the complete namespace? basic_autonomy::waypoint_generation